src/Resolver/ProductDetailImpressionDataResolver.php

```diff
@@ -63,8 +67,13 @@ private function createProductVariant(ProductVariantInterface $productVariant):
 
         $vo = $this->productDetailFactory->create();
 
+        if ($this->productIdentifier === ProductDetailInterface::ID_IDENTIFIER) {
```

Comment:
Currently the if/else here works, but it will break in case someone adds another identifier. Should we do a switch statement here and throw exception in case something unknown is set on `$this->productIdentifier`? Feels more safe to me.

Reply:
Can we also use `getProductIdentifier()` here?